src/components/App.jsx

```diff
@@ -1,24 +1,53 @@
-import { useState } from "react";
-import logo from "../images/logo.svg";
+import { useState, useEffect } from "react";
 import "./App.css";
-import Message from "./Message";
+import { MessageList } from "./MessageList";
 
 function App() {
-  const [value, setValue] = useState("");
-  function handleChange(event) {
-    setValue(event.target.value);
-  }
+  const [messageList, changeMessageList] = useState([]);
+
+  const handleAddMessage = (event) => {
+    const faker = require("faker");
+    const messageText = event.target.previousSibling.value;
+    const user = event.target.previousSibling.previousSibling.value;
+    const message_id = faker.datatype.uuid();
+    changeMessageList([...messageList, { message_id, user, messageText }]);
+  };
+
+  useEffect(() => {
+    setTimeout(() => {
```

Comment:
https://stackoverflow.com/questions/53090432/react-hooks-right-way-to-clear-timeouts-and-intervals